components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchPostCommitHook.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.search;
+
+import java.util.List;
+
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.StorageTxPostCommitHook;
+
+import com.google.common.collect.Lists;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Search TX hook.
+ *
+ * @since 3.0
+ */
+public class SearchPostCommitHook
+    extends StorageTxPostCommitHook
+{
+  private SearchFacet searchFacet;
+
+  private final List<Component> indexable;
```

Comment:
Seems like something outside the hook could be doing the accumulating, otherwise all of the hooks will need to do this sort of tracking (e.g. via a parent support class).